app/actions/handlers.py

```diff
@@ -15,6 +17,57 @@
 state_manager = IntegrationStateManager()
 
 EBIRD_API = "https://api.ebird.org/v2"
+SECONDS_IN_DAY = 86400 # 24 hours * 60 minutes * 60 seconds
+
+
+class LatestObservationDatetimeState(BaseModel):
+    latest_observation_datetime: Optional[datetime] = None
+
+    @validator('latest_observation_datetime', pre=True)
+    def clean_latest_observation_datetime(cls, v):
+        if v is None:
+            return None
+        elif isinstance(v, datetime):
+            if v.tzinfo is None:
+                return v.replace(tzinfo=timezone.utc)
+            return v.astimezone(timezone.utc)
+        else:
+            # Assuming v is a string
+            try:
+                parsed = datetime.fromisoformat(v)
+            except Exception:
+                # string is not in ISO format, return None
+                return None
+            else:
+                if parsed.tzinfo is None:
+                    parsed = parsed.replace(tzinfo=timezone.utc)
+                return parsed
```

Comment:
This got quite complicated. What i recommend is having a simple validator like this, to ensure the value is timezone aware.

```suggestion
class LatestObservationDatetimeState(BaseModel):
    latest_observation_datetime: Optional[datetime]
    @validator('latest_observation_datetime')
    def clean_latest_observation_datetime(cls, v):
        if isinstance(v, datetime):
            if v.tzinfo is None:
                return v.replace(tzinfo=timezone.utc)
        return v
```

Then when parsing data, catch validation errors just to catch unexpected cases (ex. value is a string that's not a validate date and time, or is over the wrong type).

Reply:
I had something like this in this commit:

https://github.com/PADAS/gundi-integration-ebird/pull/20/commits/27a43acd4c18e50951159d5054846a27bed81da5

So I will return to that